src/VisualStudio/CSharp/Impl/SolutionExplorer/CSharpSolutionExplorerSymbolTreeItemProvider.cs

```diff
@@ -0,0 +1,490 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using ICSharpCode.Decompiler.TypeSystem;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
+using static Microsoft.CodeAnalysis.CSharp.FindSymbols.FindSymbolsUtilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.SolutionExplorer;
+
+[ExportLanguageService(typeof(ISolutionExplorerSymbolTreeItemProvider), LanguageNames.CSharp), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpSolutionExplorerSymbolTreeItemProvider()
```

Comment:
this is the C# version of the api that, when given a node, returns the items to show for that node in solution explorer.  

Note: it operates on syntax only so we can have high perf, and not need to waste cycles on semantics.

Reply:
This type is also incredibly boring. Basically, it hits the members of interest and creates a string+glyph pair for each, in the format traditionally used for SolutionExplorerSymbols.  For example, a field is shown by `fieldName : SomeType`.

Reply:
Can we move this to Features layer and run OOP?